### PR TITLE
Add trait Stringable and class Option

### PR DESCRIPTION
**Pull request by soc**
This adds a `Stringable` trait (maybe I find a better name in the future :-)) and an implementation of an `Option` class.¹

¹ As far as I understand, we do not have support for lambdas yet?

### CODE REVIEW

File: stdlib/Option.dora [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// FIXME: Identity/Equals/Stringable constraints should all end up on the individual methods that need them, not on the class
+// After nil is gone, this class needs some more intrinsics, so that it can treat <nullptr> as an empty Option (and get rid
+// of the isPresent field). Usual caveats apply: this only works for the first "layer", not for Options inside Options, etc.
+class Option[T : Identity + Equals + Stringable](let value: T, let isPresent: Bool) {
```
Comment (dinfuehr) [inline stdlib/Option.dora]: Do you think we should implement this using subclassing? This should save some memory.

File: stdlib/Option.dora [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      return "Some(" + self.value.toString() + ")";
+    } else {
+      return "None";
+    }
+  }
+
+}
+
+// error: no ctor with definition `Option(nil, Bool)`
+// fun none[T : Identity + Equals + Stringable]() -> Option[T] = Option[T](nil, false);
```
Comment (dinfuehr) [inline stdlib/Option.dora]: While this is a bug or at least the error message super misleading, nil isn't compatible with every type. With subclassing we wouldn't have this problem...

File: stdlib/Option.dora (hunk 1 above)
Comment (soc) [reply stdlib/Option.dora]: Considering that I want to get rid of `nil`, I think I wouldn't bother with it. :-)

File: stdlib/Option.dora (hunk 2 above)
Comment (soc) [reply stdlib/Option.dora]: Dito.

**Comment by soc** [comment]
Do you have an idea how far we are away from value types?

I think they are crucial to make `Option` really useful and a prerequisite to a lot of other things.

File: stdlib/Option.dora (hunk 2 above)
Comment (dinfuehr) [reply stdlib/Option.dora]: Completely forgot about that: for these cases there is `defaultValue()`.

**Comment by dinfuehr** [comment]
I guess long enough, that changing this PR should pay off. The changes shouldn't be too dramatic. There is quite some work left until structs, ADTs and pattern matching work well enough.

File: stdlib/Option.dora (hunk 2 above)
Comment (soc) [reply stdlib/Option.dora]: Good idea! Done.

**Comment by soc** [comment]
I updated the PR to use `defaultValue`, did you have any other changes in mind?

File: tests/stringable.dora [hunk 3]
```diff
@@ -0,0 +1,13 @@
+fun main() {
+  str[Bool](true);
+  str[Bool](false);
+  str[Byte](1.toByte());
+  str[Int](1);
+  str[Long](1L);
+  // thread 'main' panicked at 'fp-register accessed as gp-register.'
```
Comment (dinfuehr) [inline tests/stringable.dora]: Is this still a bug here?

Comment (soc) [reply tests/stringable.dora]: No, shouldn't be, I'll update the commit.

**Pull request merged**